nixos/modules/services/web-apps/glitchtip.nix

```diff
@@ -0,0 +1,283 @@
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users.users = lib.mkIf (cfg.user == "glitchtip") {
+      glitchtip = {
+        description = "Glithctip service";
+        home = "/var/lib/glitchtip";
+        group = cfg.group;
+        extraGroups = lib.optionals cfg.redis.createLocally [ "redis-glitchtip" ];
+        isSystemUser = true;
+      };
+    };
+
+    users.groups = lib.mkIf (cfg.group == "glitchtip") { glitchtip = { }; };
+
+    services.redis.servers.glitchtip.enable = cfg.redis.createLocally;
+
+    services.postgresql = lib.mkIf cfg.database.createLocally {
+      enable = true;
+      ensureDatabases = [ "glitchtip" ];
+      ensureUsers = [
+        {
+          name = "glitchtip";
+          ensureDBOwnership = true;
+        }
+      ];
+    };
+
+    services.glitchtip.settings = {
+      DEBUG = lib.mkDefault 0;
+      DEBUG_TOOLBAR = lib.mkDefault 0;
+      PYTHONPATH = "${python.pkgs.makePythonPath pkg.propagatedBuildInputs}:${pkg}/lib/glitchtip";
+      DATABASE_URL = lib.mkIf cfg.database.createLocally "postgresql://@/glitchtip";
+      REDIS_URL = lib.mkIf cfg.redis.createLocally "unix://${config.services.redis.servers.glitchtip.unixSocket}";
+      CELERY_BROKER_URL = lib.mkIf cfg.redis.createLocally "redis+socket://${config.services.redis.servers.glitchtip.unixSocket}";
+    };
+
+    systemd.services =
+      let
+        commonConfig = {
+          User = cfg.user;
+          Group = cfg.group;
+          RuntimeDirectory = "glitchtip";
+          StateDirectory = "glitchtip";
+          EnvironmentFile = cfg.environmentFiles;
+
+          # hardening
+          AmbientCapabilities = "";
+          CapabilityBoundingSet = [ "" ];
+          DevicePolicy = "closed";
+          LockPersonality = true;
+          NoNewPrivileges = true;
+          ProtectKernelTunables = true;
+          ProtectProc = "invisible";
+          ProtectSystem = "strict";
+          RemoveIPC = true;
+          RestrictAddressFamilies = [
+            "AF_INET"
+            "AF_INET6"
+            "AF_UNIX"
+          ];
+          RestrictNamespaces = true;
+          RestrictRealtime = true;
+          RestrictSUIDSGID = true;
+          SystemCallArchitectures = "native";
+          PrivateDevices = true;
+          PrivateUsers = true;
+          PrivateTmp = true;
+          ProcSubset = "pid";
+          ProtectControlGroups = true;
+          ProtectHome = true;
+          ProtectHostname = true;
+          ProtectKernelLogs = true;
+          ProtectKernelModules = true;
+        };
+      in
+      {
+        glitchtip = {
+          description = "Glitchtip";
+
+          after = [
+            "network.target"
+            "redis-glitchtip.service"
+            "postgresql.service"
+          ];
+          wantedBy = [ "multi-user.target" ];
+
+          preStart = ''
+            ln -sf ${manage} /var/lib/glitchtip/glitchtip-manage
```

Comment:
We want the manage cli in the global PATH like we have for nextcloud or mastodon. Also if we did this, we should use tmpfiles